pyiceberg/table/__init__.py

```diff
@@ -630,7 +630,20 @@ def add_files(self, file_paths: List[str], snapshot_properties: Dict[str, str] =
 
         Raises:
             FileNotFoundError: If the file does not exist.
+            ValueError: Raises a ValueError given file_paths contains duplicate files
+            ValueError: Raises a ValueError given file_paths already referenced by table
         """
+        if len(file_paths) != len(set(file_paths)):
+            raise ValueError("File paths must be unique")
+
+        import pyarrow.compute as pc
+
+        expr = pc.field("file_path").isin(file_paths)
+        referenced_files = [file["file_path"] for file in self._table.inspect.files().filter(expr).to_pylist()]
+
+        if referenced_files:
+            raise ValueError(f"Cannot add files that are already referenced by table, files: {', '.join(referenced_files)}")
```

Comment:
Just one more suggestion here @amitgilad3 - should we make this behavior the default, but have a boolean flag available to disable it in the `add_files` API?

A similar [spark procedure](https://iceberg.apache.org/docs/1.5.1/spark-procedures/#add_files) has an equivalent flag `check_duplicate_files` and I feel that it could be useful because inspecting the files table requires us to read all of the active manifest files. If the user already knows that they won't be running into this issue, I think it would be useful for them to be able to disable this check.

Reply:
so my suggestion is something like:

`def add_files(self, file_paths: List[str], snapshot_properties: Dict[str, str] = EMPTY_DICT, check_duplicate_files: bool = True) -> None:`

Reply:
i also added the flag to the table api and added tests to make sure the flag works with False